engines/scc_proxy/lib/scc_proxy/engine.rb

```diff
@@ -0,0 +1,155 @@
+    end
+  end
+
+  class Engine < ::Rails::Engine
+    isolate_namespace SccProxy
+    config.generators.api_only = true
+
+    config.after_initialize do
+      # replaces RMT registration for SCC registration
+      Api::Connect::V3::Subscriptions::SystemsController.class_eval do
+        def announce_system
+          auth_header = request.headers['HTTP_AUTHORIZATION'] if request.headers.include?('HTTP_AUTHORIZATION')
+          if header_has_regcode?(auth_header)
+            # no token sent to check with SCC
+            # standard announce case
+            @system = System.create!(hostname: params[:hostname], hw_info: HwInfo.new(hw_info_params))
+          else
+            response = SccProxy.announce_system_scc(request.headers['HTTP_AUTHORIZATION'], request.request_parameters)
+            @system = System.create!(
+              login: response['login'],
+              password: response['password'],
+              hostname: params[:hostname],
+              proxy_byos: true,
+              hw_info: HwInfo.new(hw_info_params)
+            )
+          end
+          logger.info("System '#{@system.hostname}' announced")
+          respond_with(@system, serializer: ::V3::SystemSerializer, location: nil)
+        rescue *NET_HTTP_ERRORS => e
+          logger.error(
+            "Could not register system with regcode #{auth_header} " \
+              "to SCC: #{e.message}"
+          )
+          render json: { type: 'error', error: e.message }, status: status_code(e.message), location: nil
+        end
+
+        protected
+
+        def status_code(error_message)
+          error_message[0..(error_message.index(' ') - 1)].to_i
+        end
+
+        def header_has_regcode?(auth_header)
+          auth_header ||= '='
+          auth_header = auth_header[(auth_header.index('=') + 1)..-1]
+          logger.info auth_header
+          auth_header.empty?
+        end
+      end
+
+      Api::Connect::V3::Systems::ProductsController.class_eval do
+        before_action :scc_activate_product, only: %i[activate]
+
+        protected
+
+        def scc_activate_product
+          logger.info "Activating product #{@product.product_string} to SCC"
+          auth = request.headers['HTTP_AUTHORIZATION']
+          if @system.proxy_byos
+            response = SccProxy.scc_activate_product(@product, auth, params[:token], params[:email])
+            unless response.code_type == Net::HTTPCreated
+              error = JSON.parse(response.body)
+              logger.info "Could not activate #{@product.product_string}, error: #{error['error']} #{response.code}"
+              error['error'] = parse_error(error['error']) if error['error'].include? 'json'
+              raise ActionController::TranslatedError.new(error['error'])
+            end
+            logger.info "Product #{@product.product_string} successfully activated with SCC"
+            InstanceVerification.update_cache(request.remote_ip, @system.login, @product.id)
```

Comment:
Why is this needed?

Reply:
I would say so, otherwise, `ZypperAuth` would try to verify the instance _again_ ([check here](https://github.com/SUSE/rmt/blob/master/engines/zypper_auth/lib/zypper_auth/engine.rb#L17))
and we would need to do the whole process again

Reply:
What would happen after the 20 minute expiry?